extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala

```diff
@@ -36,6 +38,15 @@ class RuleApplyPermanentViewMarker extends Rule[LogicalPlan] {
     plan mapChildren {
       case p: PermanentViewMarker => p
       case permanentView: View if hasResolvedPermanentView(permanentView) =>
+        permanentView.transformAllExpressions {
+          case scalarSubquery: ScalarSubquery =>
+            scalarSubquery.plan.transformDown {
+              case p =>
+                p.setTagValue(PERMANNENT_VIEW_SUBQUERY_TAG, true)
+                p
+            }
+            scalarSubquery
+        }
```

Comment:
Since spark will optimize the scalarSubquery's plan, so here we add tag to each level like `KYUUBI_AUTHZ_TAG`.

For `InSubquery` I won't like to support in a separated pr.

Reply:
Is it efficient enough to tag the scalarSubquery.plan only?

Reply:
> I won't like to support in a separated pr?

would like?

Reply:
> > I won't like to support in a separated pr?
> 
> would like?

Yea

Reply:
> Is it efficient enough to tag the scalarSubquery.plan only?

Yea, the plan here is not optimized.

Reply:
> > Is it efficient enough to tag the scalarSubquery.plan only?
> 
> Yea, the plan here is not optimized.

What I mean here is `scalarSubquery.plan.setTagValue` only

Reply:
> > > Is it efficient enough to tag the scalarSubquery.plan only?
> > 
> > 
> > Yea, the plan here is not optimized.
> 
> What I mean here is `scalarSubquery.plan.setTagValue` only

I know, won't work

Reply:
Do you know what's happening? Tagging the whole sub-tree is a bit hypercritical to me

Reply:
> Do you know what's happening? Tagging the whole sub-tree is a bit hypercritical to me

If we only tag ScalarSubquery.plan, after generate executedPlan, may loss the top level node, since the `ScalarSubquery.plan` is not optimized.
<img width="1236" alt="截屏2023-10-16 下午2 56 45" src="https://github.com/apache/kyuubi/assets/46485123/a93905ea-8340-4204-8cc5-ddf0844dd487">

Reply:
PlanSubqueries seem to happen in the planner not in the optimizer. Do you mean the execution.ScalarSubquery is already in analyzer instead of a logical one?

Reply:
> PlanSubqueries seem to happen in the planner not in the optimizer. Do you mean the execution.ScalarSubquery is already in analyzer instead of a logical one?

<img width="1094" alt="截屏2023-10-16 下午3 28 24" src="https://github.com/apache/kyuubi/assets/46485123/6f440d1c-1664-4e81-b782-b705b3ca48ab">

Reply:
@yaooqinn Change to add a new node to do this. Pls take a look again